pkg/apis/ndm/v1alpha1/types.go

```diff
@@ -71,3 +71,47 @@ type StoragePoolList struct {
 
 	Items []StoragePool `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=disk
+
+// Disk describes disk resource.
+type Disk struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata, omitempty"`
+
+	Spec DiskSpec `json:"spec"`
+}
+
+// DiskSpec is the specification for the disk stored as CRD
+type DiskSpec struct {
+	Uuid       string `json:"uuid"`         //unique identifier for the disk
```

Comment:
Whats this UUID based on? can we use the disk GUID SPC4, section 7.

Reply:
we can discuss this on the design. Please refer #1